pkg/apis/core/types_shoot.go

```diff
@@ -301,6 +303,46 @@ type NginxIngress struct {
 	ExternalTrafficPolicy *corev1.ServiceExternalTrafficPolicyType
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// ShootControlPlane relevant types                                                             //
+//////////////////////////////////////////////////////////////////////////////////////////////////
+
+// ShootControlPlane holds information about the general settings for the control plane of a shoot.
+type ShootControlPlane struct {
+	// HighAvailability holds the configuration settings for high availability of the
+	// control plane of a shoot.
+	HighAvailability HighAvailability
```

Comment:
Shouldn't this be optional, i.e. a pointer type?

Reply:
For now we have only kept `ControlPlane` optional. If it is defined then `HighAvailability` also needs to be defined. If tomorrow additional fields are added to `ControlPlane` then we can make `HighAvailability` a pointer type.